File: was/auth/src/main/java/org/example/domain/entity/MemberReason.java
package org.example.domain.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.example.domain.enums.UsingReason;

import java.util.UUID;

@Entity
@Table(name = "MemberReason")
@NoArgsConstructor
@Getter
public class MemberReason {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private UUID memberId;

    @Column(nullable = false)
    @Enumerated(value = EnumType.STRING)
    private UsingReason reason;

    protected MemberReason(UUID id, UUID memberId, UsingReason reason) {
        this.memberId = memberId;
        this.reason = reason;
        this.id = id;
    }

    public MemberReason(UUID memberId, UsingReason reason) {
        this.memberId = memberId;
        this.reason = reason;
    }
}

File: was/member/src/main/java/org/example/domain/entity/MemberReason.kt
package org.example.domain.entity

import jakarta.persistence.*
import org.example.domain.enums.UsingReason
import java.util.UUID

@Entity
@Table(name = "MemberReason")
open class MemberReason{

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    val id: UUID?

    @Column(nullable = false)
    val memberId: UUID

    @Column(nullable = false)
    @Enumerated(value = EnumType.STRING)
    val reason: UsingReason

    constructor(id: UUID?, memberId: UUID, reason: UsingReason) {
        this.memberId = memberId
        this.reason = reason
        this.id = id
    }

    protected constructor() : this(null, UUID.randomUUID(), UsingReason.DIET) {}
}